### PR TITLE
Feature/us11 notes

### PR DESCRIPTION
**Pull request by ufelgen**
following [this US](https://github.com/ufelgen/capstone-project/issues/18) I equipped the popup menu of the vocab cards with an option to add notes. these can also be edited. I coded notes for the first word in the animals category for demonstration purposes.

### CODE REVIEW

File: components/SingleWordHeading/SingleWordHeading.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {query1.translation} {query1.gender && `(${query1.gender})`}
+      </h2>
+    </StyledHeadingWrapper>
+  );
+}
+
+const StyledHeadingWrapper = styled.section`
+  margin: 0.625rem 0.625rem 1.25rem 0.625rem;
+  background-color: White;
+  color: darkmagenta;
```
Comment (flipkoehler) [inline components/SingleWordHeading/SingleWordHeading.js:19]: a very cool new component 👍 
just an idea - nothing you have to change. Maybe you can place your colors to the globalStyles file. Than you can reference to it - e.g. "var(--darkmagenta)"

File: components/DeclensionForm/DeclensionForm.js [hunk 2]
```diff
@@ -38,14 +39,7 @@ export default function DeclensionForm({ currentWord, onAddDeclensionForm }) {
...
-        <h2>{query1.flag}</h2>
-        <h2>{base.translation}</h2>
-        <h2>
-          {query1.translation} ({query1.gender})
-        </h2>
-      </StyledHeadingWrapper>
+      <SingleWordHeading base={base} query1={query1} />
       <StyledForm onSubmit={handleSubmitDeclensionForm}>
         <StyledSpecificationWrapper>
           <label htmlFor="specification">declension type:</label>
```
Comment (flipkoehler) [inline components/DeclensionForm/DeclensionForm.js:45]: the code in line 78 - 87 isn't used anywhere else. You can remove it.

File: pages/notes/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Link from "next/link";
```
Comment (flipkoehler) [inline pages/notes/index.js:1]: I don't understand this page - but its working fine 😄

File: components/NotesForm/NotesForm.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import styled from "styled-components";
+import { StyledForm } from "../StyledForm";
+
+export default function NotesForm({
+  currentWord,
+  onSaveNotes,
+  inputLabel,
+  buttonLabel,
+  editing,
+  editId,
```
Comment (flipkoehler) [inline components/NotesForm/NotesForm.js:10]: this one isn't used in the code

File: components/PopupMenu/PopupMenu.js [hunk 5]
```diff
@@ -19,11 +19,18 @@ export default function PopupMenu({
...
-        </Link>
+        <>
+          <Link href={`/declension/${id}`}>
+            <StyledMenuButton>
+              {prop.query1?.declension ? "declension" : "+ declension"}
+            </StyledMenuButton>
+          </Link>
+          <Link href={`/notes/${id}`}>
+            <StyledMenuButton>
+              {prop.notes ? "notes" : "+ notes"}
```
Comment (flipkoehler) [inline components/PopupMenu/PopupMenu.js:30]: i love the little extra UX with the "+" before the word❤️

File: components/StyledIcons.js [hunk 6]
```diff
@@ -1,10 +1,14 @@
 import styled from "styled-components";
 
 export function DeclensionIcon() {
-  return <StyledIcon>D</StyledIcon>;
+  return <StyledDeclensionIcon>D</StyledDeclensionIcon>;
```
Comment (flipkoehler) [inline components/StyledIcons.js:4]: nice idea!

**Review by flipkoehler — APPROVED** [review]
just a minor thing to change, which isn't even part of this PR. 
Very nice work Ulli! You are passing the holy s%$& out of that props! Keep up the good work 😊

File: components/PopupMenu/PopupMenu.js (hunk 5 above)
Comment (AntjeWalter) [reply components/PopupMenu/PopupMenu.js:30]: I totally agree :)

**Review by AntjeWalter — APPROVED** [review]
Nothing to add, works perfectly fine for me! :)

**Pull request merged**